src/pages/publication/contributors_tab/components/AffiliationsCell.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, useState } from 'react';
+import { Institution } from '../../../../types/contributor.types';
+import { Button } from '@material-ui/core';
+import SelectInstitution from '../../../../components/SelectInstitution';
+import Modal from '../../../../components/Modal';
+import { useFormikContext, FormikProps } from 'formik';
+import { FormikPublication, BackendTypeNames } from '../../../../types/publication.types';
+import { SpecificContributorFieldNames } from '../../../../types/publicationFieldNames';
+import { FormikInstitutionUnit } from '../../../../types/institution.types';
+import { useTranslation } from 'react-i18next';
+import NormalText from '../../../../components/NormalText';
+import styled from 'styled-components';
+import ConfirmDialog from '../../../../components/ConfirmDialog';
+import AddIcon from '@material-ui/icons/Add';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { publicationLanguages, LanguageCodes } from '../../../../types/language.types';
+
+const StyledAffiliationsCell = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const StyledRemoveAffiliationButton = styled(Button)`
+  margin-left: 1rem;
+`;
+
+interface AffiliationsCellProps {
+  affiliations: Institution[];
+  baseFieldName: string;
+  contributorName: string;
+}
+
+const AffiliationsCell: FC<AffiliationsCellProps> = ({ affiliations, baseFieldName, contributorName }) => {
+  const { t } = useTranslation('publication');
+  const [openAffiliationModal, setOpenAffiliationModal] = useState(false);
+  const [affiliationToRemove, setAffiliationToRemove] = useState<Institution | null>(null);
+  const { values, setFieldValue }: FormikProps<FormikPublication> = useFormikContext();
+
+  const toggleAffiliationModal = () => setOpenAffiliationModal(!openAffiliationModal);
+
+  const addAffiliation = (value: FormikInstitutionUnit) => {
+    // TODO: Set hierarchy in state? get from backend?
+    const mostSpecificUnit = value.subunits.pop() ?? value;
+    const labelKey =
+      publicationLanguages.find((language) => language.value === values.entityDescription.language)?.id ??
+      LanguageCodes.ENGLISH;
+
+    const addedAffiliation: Institution = {
+      type: BackendTypeNames.ORGANIZATION,
+      id: mostSpecificUnit.id,
+      labels: {
+        [labelKey]: mostSpecificUnit.name,
```

Comment:
Har luftet dette for @brinxmat og @axthosarouris :
Vi trenger egentlig ikke forholde oss så mye til labels, men vi kan bruke innholdets hovedspråk som utgangspunkt for key